qa/sc_evm_rpc_block_tags.py

```diff
@@ -0,0 +1,192 @@
+                assert_true(resBlockTxCount['result'] == kwargs['BlockTransactionCount'])
+                resBlock = resBlock['result']
+                if tag == 'earliest':
+                    assert_true(resBlock['parentHash'] == kwargs['Block'])
+                else:
+                    assert_true(resBlock['number'] == kwargs['Block'])
+            else:
+                assert_true(resBalance['error']['code'] == kwargs['Balance'])
+                assert_true(resTxCount['error']['code'] == kwargs['TransactionCount'])
+                assert_true(resBlockTxCount['error']['code'] == kwargs['BlockTransactionCount'])
+                assert_true(resBlock['error']['code'] == kwargs['Block'])
+
+        if tag in ['pending', 'latest'] and secondAddress is not None:
+            self.__do_contract_call_tests(self.sc_nodes[0], tag, secondAddress)
+
+    def run_test(self):
+        sc_node_1 = self.sc_nodes[0]
+        sc_node_2 = self.sc_nodes[1]
+
+        ft_amount_in_zen = Decimal('500.0')
+        ft_amount_in_wei_hex_str = str(hex(convertZenToWei(ft_amount_in_zen)))
+
+        self.sc_ac_setup(ft_amount_in_zen=ft_amount_in_zen)
+
+        evm_address_sc1 = remove_0x_prefix(self.evm_address)
+        evm_address_sc2 = sc_node_1.wallet_createPrivateKeySecp256k1()["result"]["proposition"]["address"]
+
+        # create an address on sc_node_2 for nonempty storage
+        sc_node_2.wallet_createPrivateKeySecp256k1()["result"]["proposition"]["address"]
+
+        self.sc_sync_all()
+
+        expResp = {'Balance': '0x0', 'TransactionCount': '0x0', 'BlockTransactionCount': '0x0',
+                   'Block': '0x0000000000000000000000000000000000000000000000000000000000000000'}
+        self.__send_and_assert_tag_rpc_methods('earliest', **expResp)
+
+        expResp['Balance'], expResp['Block'] = ft_amount_in_wei_hex_str, '0x3'
+        self.__send_and_assert_tag_rpc_methods('pending', **expResp)
+
+        expResp['Block'] = '0x2'
+        self.__send_and_assert_tag_rpc_methods('latest', **expResp)
+
+        # safe / finalized will return -39001 Unknown Block error until we have 100 sidechain blocks
+        expResp = {k: -39001 for k in expResp}
+        self.__send_and_assert_tag_rpc_methods('safe', **expResp)
+        self.__send_and_assert_tag_rpc_methods('finalized', **expResp)
+
+        # Create some TX in mempool
+        nonce_addr_1 = 0
+        common_tx_list = []
+        for i in range(4):
+            common_tx_list.append(
+                createEIP1559Transaction(sc_node_1, fromAddress=evm_address_sc1, toAddress=evm_address_sc2,
+                                         nonce=nonce_addr_1, gasLimit=230000, maxPriorityFeePerGas=900000000,
+                                         maxFeePerGas=900000000, value=1))
+            nonce_addr_1 += 1
+
+        # Check pending state and block functions
+        expResp = {'Balance': '0x1b1ae49220f02adffc', 'TransactionCount': '0x4', 'BlockTransactionCount': '0x4',
```

Comment:
please add the comment explaining the value `0x1b1ae49220f02adffc`
what is it decimal form
and why do we expect exactly this value.

Reply:
I refactored this a bit to be a lot more generic and understandable